packages/forms-web-app/src/controllers/selected-appeal/appeal-details/index.js

```diff
@@ -8,51 +8,57 @@ const { rows } = require('./appeal-details-rows');
 const { getLPAUserFromSession } = require('../../../services/lpa-user.service');
 
 /**
- * @type {import('express').Handler}
+ * Shared controller for /appeals/:caseRef/appeal-details, manage-appeals/:caseRef/appeal-details rule-6-appeals/:caseRef/appeal-details
+ * @param {string} layoutTemplate - njk template to extend
+ * @returns {import('express').RequestHandler}
  */
-exports.get = async (req, res) => {
-	const appealNumber = req.params.appealNumber;
-	const userRouteUrl = req.originalUrl;
+exports.get = (layoutTemplate = 'layouts/no-banner-link/main.njk') => {
```

Comment:
I like this solution. 

For the questionnaire details page I was determining the correct template based on the userType further down, doing it by the actual route is much less likely to show the wrong template